File: src/main/java/dmgcalculator/entities/Range.java
package dmgcalculator.entities;

public class Range {

    private int min;
    private int max;

    public Range() {
        this(0);
    }

    public Range(int value) {
        this(value, value);
    }

    public Range(int min, int max) {
        this.min = min;
        this.max = max;
    }

    public void set(int min, int max) {
        this.min = min;
        this.max = max;
    }

    public void set(int value) {
        this.set(value, value);
    }

    public int getMin() {
        return min;
    }

    public int getMax() {
        return max;
    }

    public int getValue() {
        return max;
    }

    public boolean isConstantsValue() {
        return min == max;
    }
}

File: src/main/kotlin/dmgcalculator/entities/Range.kt
package dmgcalculator.entities

data class Range(var min: Int, var max: Int) {

    val value: Int
        get() = max

    val isConstantsValue: Boolean
        get() = min == max

    constructor(value: Int = 0) : this(value, value)

    constructor(other: Range) : this(other.min, other.max)

    fun set(min: Int, max: Int) {
        this.min = min
        this.max = max
    }

    fun set(value: Int) {
        this.set(value, value)
    }

    fun set(other: Range) {
        this.set(other.min, other.max)
    }

    operator fun plus(amount: Int): Range {
        return Range(min + amount, max + amount)
    }

    operator fun plus(other: Range): Range {
        return Range(min + other.min, max + other.max)
    }

    operator fun minus(amount: Int): Range {
        return Range(min - amount, max - amount)
    }

    operator fun minus(other: Range): Range {
        return Range(min - other.max, max - other.min)
    }

    operator fun minusAssign(other: Range) {
        min -= other.max
        max -= other.min
    }

    operator fun plusAssign(other: Range) {
        min += other.min
        max += other.max
    }
}